### ISSUE TITLE
Embedded Tomcat in spring-boot not able a create multiple session for multiple login request

### ISSUE THREAD

**Issue by subha045** [spring-projects/spring-boot#9266]
I am creating a file check in utility using spring-boot. In which I am exposing a URL like(http://localhost:8181/batchDriver?(....parameters....)). This URL will be invoked form batch. Tomcat will be running on port 8181 and we are going to use the same instance of the tomcat for each batch.

Each time when the link will be invoked the spring boot application will perform the below steps - 1. Login to Oracle WebCenter Imaging using imaging-client.jar. 2. After login it will associate the parameters received from the URL with a file. 3. Call Imaging document service and upload the file. 4. Log out from Oracle WebCenter Imaging.

Problem Statement When we are using single host URL to the Imaging webservice URL it is working fine for single batch and multiple concurrent batches. But when I am using the load balanced version of the Imaging webservice URL. It's working fine for single batch but not working for multiple concurrent batches. Below is the exception we are getting - 1. javax.xml.ws.soap.SOAPFaultException: InvalidSecurity : error in processing the WS-Security security header 2. oracle.imaging.ImagingException: TCM-00009: No session has been established.

What I think the embedded tomcat is storing the session for first login with one manage server but when the manage server changed then it's not creating a new session again for that manage server. As for load balancing URL it's going to get dynamic manage servers. For single host/manage server it's not giving any problem.

How to create a new session for each request in embedded tomcat of force it to not store any session. Please also suggest if any other way to solve the issue.

**Comment by wilkinsona** [comment]
Sorry, I'm finding it rather hard to follow what you are trying to do. Could you please creat a minimal sample that focusses on the HTTP session creation which, I think, is the root of the problem you are having.

**Comment by subha045** [comment]
Sorry for the delayed response. Please find the code snippet below -

package devguidesamples;
import java.util.List;
import java.util.Locale;
import oracle.imaging.Application;
import oracle.imaging.ApplicationService;
import oracle.imaging.BasicUserToken;
import oracle.imaging.ImagingException;
import oracle.imaging.NameId;
import oracle.imaging.ServicesFactory;
import oracle.imaging.UserToken;
public class IntroSample {
 public static void main(String[] args) {
 try { // try-catch
 UserToken credentials = new BasicUserToken("ipmuser", "ipmuserpwd");
 ServicesFactory servicesFactory =
 ServicesFactory.login(credentials, Locale.US,
"http://ipmhost:16000/imaging/ws");
 try { // try-finally to ensure logout
 ApplicationService appService =
servicesFactory.getApplicationService();

 // List the viewable applications to confirm that "Invoices" exists
 List<NameId> appsList =
appService.listApplications(Application.Ability.VIEW);
 for (NameId appNameId: appsList) {
 System.out.println( appNameId );
 }
 }
 finally {
 if (servicesFactory != null) {
servicesFactory.logout();
 }
 }
 }
 catch (ImagingException e) {
 System.out.println(e.getMessage());
 }
 }
}

The problem is happening in line - 
ServicesFactory servicesFactory = ServicesFactory.login(credentials, Locale.US, "http://ipmhost:16000/imaging/ws");
when we are using load balanced URL instead of http://ipmhost:16000/imaging/ws then the concurrent login request is failing due to below exceptions.
1. javax.xml.ws.soap.SOAPFaultException: InvalidSecurity : error in processing the WS-Security security header 
2. oracle.imaging.ImagingException: TCM-00009: No session has been established.

But if we use http://ipmhost:16000/imaging/ws URL then it's not giving any exception in time of concurrent login request.

**Comment by wilkinsona** [comment]
Thanks, but that's not really what I'm looking for. Unfortunately, I don't have the time to piece together a working sample from a (poorly formatted) code snippet. If you would like me to spend some time trying to help you diagnose your problem, please spend some time providing a minimal, yet complete example of the problem; something that I can unzip or git clone and then run would be ideal.

**Comment by subha045** [comment]
Sorry but I can not provide that as it's client confidential and the jars are also licensed by Oracle.

**Comment by wilkinsona** [comment]
It doesn't have to be your client's code or involve any of Oracle's code. It just needs to be something that reproduces the problem you're seeing, i.e. the inability to create multiple sessions from multiple logins.

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by spring-issuemaster** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.
